File: lib/src/models/DestinyItemSocketEntryPlugItemRandomizedDefinition.cs
using System.Runtime.Serialization;

namespace BungieNetApi.Model {
	[DataContract]
	public class DestinyItemSocketEntryPlugItemRandomizedDefinition{

		/// <summary>
		/// The hash identifier of a DestinyInventoryItemDefinition representing the plug that can be inserted.
		/// </summary>
		[DataMember(Name="plugItemHash", EmitDefaultValue=false)]
		public uint PlugItemHash { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as DestinyItemSocketEntryPlugItemRandomizedDefinition);
        }

		public bool Equals(DestinyItemSocketEntryPlugItemRandomizedDefinition input)
		{
			if (input == null) return false;

			return
				(
                    PlugItemHash == input.PlugItemHash ||
                    (PlugItemHash != null && PlugItemHash.Equals(input.PlugItemHash))
                ) ;
		}

		/* 
		public override int GetHashCode()
		{
			unchecked // Overflow is fine, just wrap
			{
				int hashCode = 41;
				hashCode = hashCode * 59 + this.PlugItemHash.GetHashCode();
				return hashCode;
			}
		}*/
	}
}

File: pkg/models/DestinyItemSocketEntryPlugItemRandomizedDefinition.go
package bungieapigo

type DestinyItemSocketEntryPlugItemRandomizedDefinition struct {

	// Indicates if the plug can be rolled on the current version of the item. For example, older
	// versions of weapons may have plug rolls that are no longer possible on the current versions.
	CurrentlyCanRoll bool `json:"currentlyCanRoll"`

	// The hash identifier of a DestinyInventoryItemDefinition representing the plug that can be
	// inserted.
	PlugItemHash int `json:"plugItemHash"`
}
